plugins/tiddlywiki/multiwikiserver/modules/routes/handlers/post-bag.js

```diff
@@ -13,42 +13,70 @@ description
 \*/
 (function() {
 
-/*jslint node: true, browser: true */
-/*global $tw: false */
-"use strict";
-
-exports.method = "POST";
-
-exports.path = /^\/bags$/;
-
-exports.bodyFormat = "www-form-urlencoded";
-
-exports.csrfDisable = true;
-
-exports.useACL = true;
-
-exports.entityName = "bag"
-
-exports.handler = function(request,response,state) {
-	if(state.data.bag_name) {
-		const result = $tw.mws.store.createBag(state.data.bag_name,state.data.description);
-		if(!result) {
-			state.sendResponse(302,{
-				"Content-Type": "text/plain",
-				"Location": "/"
-			});
+	/*jslint node: true, browser: true */
+	/*global $tw: false */
+	"use strict";
+	
+	exports.method = "POST";
+	
+	exports.path = /^\/bags$/;
+	
+	exports.bodyFormat = "www-form-urlencoded";
+	
+	exports.csrfDisable = true;
+	
+	exports.useACL = true;
+	
+	exports.entityName = "bag"
+	
+	exports.handler = function(request,response,state) {
+		var server = state.server,
+			sqlTiddlerDatabase = server.sqlTiddlerDatabase;
+		
+		// Handle DELETE request
+		if(state.data._method === "DELETE") {
```

Comment:
I am not keen on the way that the DELETE /bags/:bag-name method is piggybacked into the same handler as the POST /bags method.

I'd prefer to implement DELETE /bags/:bag-name as a separate handler, and to move the logic that allows POST as a synonym for DELETE into mws-server.js (similarly to the PUT/POST mapping it already does). This would allow REST purists to do things properly if they want to.